### PR TITLE
Add SitemapBuildableView

### LINKED ISSUE #37: Add SitemapBuildableView
Sites need a sitemap and it'd be great to be able to generate this as part of the static build process.

I'm currently putting up a PR to answer this issue.

### PR DESCRIPTION
**Pull request by loicteixeira**
Rebase & continue #38. Fix #37.

### CODE REVIEW

**Comment by loicteixeira** [comment]
Currently, sitemaps are build at the correct location for single and multisites, however the content is incorrect for multisites.

Despite creating a request which is multisites aware, and that request being passed to the Sitemap object, it then [use the paginator](https://github.com/wagtail/wagtail/blob/16bf4cac56684401734201c9bf2aa7d9ce583ac3/wagtail/contrib/sitemaps/sitemap_generator.py#L40) which doesn't seem to be multisite aware and return pages of the default site. There's `self.items` which would be site aware but isn't used? So it could possibly be an upstream issue? Or I'm just all mixed up in the sitemap logic.

I'll leave this aside for a bit so I can focus on releasing a new version of the package and then come back to this.

File: src/wagtailbakery/views.py [hunk 1]
```diff
@@ -140,3 +141,37 @@ class AllPublishedPagesView(AllPagesView):
     def get_queryset(self):
         pages = super(AllPublishedPagesView, self).get_queryset()
         return pages.live()
+
+
+class SitemapBuildableView(BuildableMixin):
+
```
Comment (tomdyson) [inline src/wagtailbakery/views.py:147]: Can we have a comment to explain the function of this view?

File: tests/integration/test_views.py [hunk 2]
```diff
@@ -91,3 +92,66 @@ def test_all_pages_for_single_page(page):
...
     assert qs.filter(id=page.id).exists()
+
+
+@pytest.mark.django_db
+def test_sitemap_content_for_single_site(settings, site):
+    view = SitemapBuildableView()
+    view.request = view.create_request(view.sitemap_path)
+
+    content = view.get_content().decode()
+    expected_content = (
```
Comment (tomdyson) [inline tests/integration/test_views.py:103]: This test looks fragile. Should we assume that the exact formatting of Django's sitemap generation will be consistent across versions? As an alternative we could parse the XML and check for the namespace, a single `<loc>` element inside a single `<url>` element etc.

**Review by tomdyson — COMMENTED** [review]
Thanks @loicteixeira, looks good!

We'll need some basic documentation, e.g. in the `Configuration` section of the README.

**Pull request closed without merging**